### ISSUE TITLE
Bond paths without enough points and terminating too far from nuclear CPs

### ISSUE THREAD

**Issue by twilsonco** [MolecularTheoryGroup/BondalyzerWindows#2]
![image](https://user-images.githubusercontent.com/7284371/40740081-16fa477c-6405-11e8-90be-45cda73ba67f.png)

this is Fulv66_CAMYB3LYP_TZP_c2v_s0_opt1_1a1_MOs.lay in StorageWorkspace.

**Comment by twilsonco** [comment]
GPs and GBs look ok
![image](https://user-images.githubusercontent.com/7284371/40740258-9718a34a-6405-11e8-9169-6218c4c55c43.png)

**Comment by twilsonco** [comment]
Solved.
There were in fact GPs in GBA that were prematurely terminating. 
The problem was that, at points, rho can be stiff (as in a stiff system of ODEs).
I changed the ODE solver from Runge-Kutta to the [Implicit Bulirsch-Stoer method of Bader and Deuflhard](https://www.gnu.org/software/gsl/doc/html/ode-initval.html), which is good for stiff problems, seems to work well elsewhere in the density (so we shouldn't need to switch method dynamically), and required that I provide the Jacobian (i.e. Hessian in the density) as well, which I did numerically, so it slowed down the GP algorithm a bit.
